cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/ExecutorStatusTest.java

```diff
@@ -0,0 +1,123 @@
+    }
+
+    @Test
+    public void testGetTotalDataToMove() {
+        ExecutorStatus es0 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, "10000", DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es0.getTotalDataToMove(), is(10000));
+
+        // Test missing field value is zero
+        ExecutorStatus es1 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, "", DEFAULT_TRIGGERED_TASK_REASON));
+        assertThat(es1.getTotalDataToMove(), is(0));
+
+        // Test missing field throws IllegalArgumentException
+        assertThrows(IllegalArgumentException.class, () ->
+                new ExecutorStatus(createExecutorStatusJson(null, null, null, null)));
+    }
+
+    @Test
+    public void testGetTaskStartTime() {
+        ExecutorStatus es0 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE,
+                "No reason provided (Client: 172.17.0.1, Date: 2024-11-15T19:41:27Z)"));
+        assertThat(es0.getTaskStartTime().toString(), is("2024-11-15T19:41:27Z"));
+
+        ExecutorStatus es1 = new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE,
+                "(Client: 172.17.0.1, Date: 2024-11-10T23:25:27Z)"));
+        assertThat(es1.getTaskStartTime().toString(), is("2024-11-10T23:25:27Z"));
+
+        // Test missing date-string fails
+        assertThrows(IllegalArgumentException.class, () -> new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE, "")));
+
+        // Test date-string in a non-UTC timezone fails
+        assertThrows(IllegalArgumentException.class, () -> new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE,
+                "No reason provided (Client: 172.17.0.1, Date: 2024-11-15T20:41:27+01:00)")));
+
+        // Test malformed date-string fails
+        assertThrows(IllegalArgumentException.class, () -> new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE,
+                "No reason provided (Client: 172.17.0.1, Date: 2024-11-15T20:41:2")));
+
+        // Test missing date-string fails
+        assertThrows(IllegalArgumentException.class, () -> new ExecutorStatus(createExecutorStatusJson(
+                DEFAULT_STATE, DEFAULT_FINISHED_DATA_MOVEMENT, DEFAULT_TOTAL_DATA_TO_MOVE,
+                "No reason provided (Client: 172.17.0.1)")));
+
+        // Test missing field throws IllegalArgumentException
+        assertThrows(IllegalArgumentException.class, () ->
+                new ExecutorStatus(createExecutorStatusJson(null, null, null, null)));
```

Comment:
There's already the exact same assertion on line 84:85

Reply:
Also same issue it's not actually checking the `triggeredTaskReason` field.

Reply:
Fixed.